src/main/scala/mesosphere/marathon/core/task/termination/TaskKillService.scala

```diff
@@ -0,0 +1,53 @@
+package mesosphere.marathon.core.task.termination
+
+import akka.Done
+import mesosphere.marathon.core.task.Task
+
+import scala.concurrent.Future
+
+/**
+  * A service that handles killing tasks. This will take care about extra logic for lost tasks,
+  * apply a retry strategy and throttle kill requests to Mesos.
+  */
+trait TaskKillService {
+  /**
+    * Kill the given tasks and return a future that is completed when all of the tasks
+    * have been reported as terminal.
+    *
+    * @param tasks the tasks that shall be killed.
+    * @param reason the reason why the task shall be killed.
+    * @return a future that is completed when all tasks are killed.
+    */
+  def killTasks(tasks: Iterable[Task], reason: TaskKillReason): Future[Done]
+
+  /**
+    * Kill a task by the given Id. The implementation should add the task onto a queue that is processed
+    * short term and will eventually kill the task.
+    *
+    * @param taskId the id of the task that shall be killed.
+    * @param reason the reason why the task shall be killed.
+    * @return a future that is completed when all tasks are killed.
+    */
+  // TODO: remove function and only allow killing unknown tasks or Tasks
```

Comment:
I chose to follow the _right_ path, removed the function and adjusted code in the TaskReplaceActor so this now uses `killTask` instead of the removed function.